src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +430,1174 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
             }
         }
 
-        private static string[] GetNullableDataKeysAsStrings(ConcurrentDictionary<object, int> nullableData) =>
-            nullableData.Keys.Select(key => GetNullableDataKeyAsString(key)).OrderBy(key => key).ToArray();
+        private readonly struct NullableDirectives
+        {
+            internal readonly string[] Directives;
+            internal readonly NullableContextState.State ExpectedWarningsState;
+            internal readonly NullableContextState.State ExpectedAnnotationsState;
+
+            internal NullableDirectives(string[] directives, NullableContextState.State expectedWarningsState, NullableContextState.State expectedAnnotationsState)
+            {
+                Directives = directives;
+                ExpectedWarningsState = expectedWarningsState;
+                ExpectedAnnotationsState = expectedAnnotationsState;
+            }
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_01()
+        {
+            var projectSettings = new[]
+            {
+                (NullableContextOptions?)null,
+                NullableContextOptions.Disable,
+                NullableContextOptions.Warnings,
+                NullableContextOptions.Annotations,
+                NullableContextOptions.Enable,
+            };
+            var nullableDirectives = new[]
+            {
+                new NullableDirectives(new string[0], NullableContextState.State.Unknown, NullableContextState.State.Unknown),
+                new NullableDirectives(new[] { "#nullable disable" }, NullableContextState.State.Disabled, NullableContextState.State.Disabled),
+                new NullableDirectives(new[] { "#nullable enable" }, NullableContextState.State.Enabled, NullableContextState.State.Enabled),
+                new NullableDirectives(new[] { "#nullable restore" }, NullableContextState.State.ExplicitlyRestored, NullableContextState.State.ExplicitlyRestored),
+                new NullableDirectives(new[] { "#nullable disable annotations" }, NullableContextState.State.Unknown, NullableContextState.State.Disabled),
+                new NullableDirectives(new[] { "#nullable disable annotations", "#nullable enable warnings" }, NullableContextState.State.Enabled, NullableContextState.State.Disabled),
```

Comment:
> "#nullable enable warnings" [](start = 79, length = 28)

As warnings come first in the expected section, consider putting warnings here first as well. Make it a little less mind-bending to read.